src/actions/preferences.js

```diff
@@ -99,32 +99,55 @@ export function makeGroupMessageVisibleIfNecessary(channelId: string): ActionFun
             }
 
             getProfilesInChannel(channelId, 0)(dispatch, getState);
-            savePreferences(currentUserId, [preference])(dispatch);
+            savePreferences(currentUserId, [preference])(dispatch, getState);
         }
 
         return {data: true};
     };
 }
 
-export function savePreferences(userId: string, preferences: Array<PreferenceType>) {
+export function savePreferences(userId: string, preferences: Array<PreferenceType>): ActionFunc {
     return async (dispatch: DispatchFunc) => {
-        dispatch({
-            type: PreferenceTypes.RECEIVED_PREFERENCES,
-            data: preferences,
-            meta: {
-                offline: {
-                    effect: () => Client4.savePreferences(userId, preferences),
-                    commit: {
-                        type: PreferenceTypes.RECEIVED_PREFERENCES,
-                    },
-                    rollback: {
-                        type: PreferenceTypes.DELETED_PREFERENCES,
-                        data: preferences,
-                    },
-                },
-            },
-        });
+        dispatchPrefSave(dispatch, userId, preferences);
+        return {data: true};
+    };
+}
+
+export function saveCurrentUserPreferences(preferences: Array<PreferenceType>): ActionFunc {
```

Comment:
Is this action needed here? This can be done directly in the webapp, and the call savePreferences

Reply:
Yes totally possible, but in that case components should know `userId`

Reply:
Compinent do not need to know the userId, you can add an action in the webapp that gets the userId from the store and then calls the savePreferences action, no need to have it here

Reply:
For me it seems to be a bit declarative way, `saveCurrentUserPreferences()` you know that it updates current users prefs comparing to `savePreferences()`

Reply:
Yes agree, but this action can live only in the webapp, no need to have it here

Reply:
I’m just looking to have a clean code base and avoid client specific actions as much as possible.